### PR TITLE
update aws deployment for 2.6.0

### PR DESCRIPTION
**Pull request by kristi**
### Motivation
I tried to follow the Aws deployment guide at https://pulsar.apache.org/docs/en/deploy-aws/ but found it was pretty outdated - It was trying to install pulsar 2.1.0-incubating. This PR updates it to install 2.6.0.

### Modifications

* Updated the pulsar version to 2.6.0
  * Fixed download location for 2.6.0
  * Updated config files for 2.6.0
  * Fixed connector installation for 2.6.0
  * Fixed Ansible's yum warning about installing multiple packages

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change does not change any code

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: only the aws tutorial deployment

### Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

already documented at https://pulsar.apache.org/docs/en/deploy-aws/

### CODE REVIEW

File: deployment/terraform-ansible/deploy-pulsar.yaml [hunk 1]
```diff
@@ -123,12 +124,45 @@
...
+        path: "/opt/pulsar/{{ item }}"
+        state: directory
+      loop:
+        - connectors
+    - name: Download Pulsar IO packages
+      get_url:
+        url: https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=pulsar/pulsar-{{ pulsar_version }}/connectors/pulsar-io-{{ item }}-{{ pulsar_version }}.nar
+        dest: /opt/pulsar/connectors/pulsar-io-{{ item }}-{{ pulsar_version }}.nar
+      loop:
+#        - aerospike
```
Comment (merlimat) [inline deployment/terraform-ansible/deploy-pulsar.yaml:138]: Should these go uncommented by default, to keep it consistent with previous behavior?

Comment (kristi) [reply deployment/terraform-ansible/deploy-pulsar.yaml:138]: I'm not sure that the previous setup was widely used, given how out of date it was. I feel like it's more of a tutorial / jumping off point, than something that someone would use regularly and be confused by its behavior.

It's just a lot of time to download all of them.  Feels a bit more friendly to only download a couple, given that these are optional and that people using this are probably trying it out for the first time.

Comment (kristi) [reply deployment/terraform-ansible/deploy-pulsar.yaml:138]: I added a note in the documentation page about the connectors

File: deployment/terraform-ansible/deploy-pulsar.yaml [hunk 2]
```diff
@@ -28,20 +28,21 @@
...
-        http_url: "http://{{ hostvars[groups['proxy'][0]].public_ip }}:8080/"
-        pulsar_version: "2.1.0-incubating"
+        zookeeper_servers: "{{ groups['zookeeper']|map('extract', hostvars, ['ansible_default_ipv4', 'address'])|map('regex_replace', '^(.*)$', '\\1:2181') | join(',') }}"
+        service_url: "{{ pulsar_service_url }}"
+        http_url: "{{ pulsar_web_url }}"
+        pulsar_version: "2.6.0"
     - name: Download Pulsar binary package
       unarchive:
-        src: http://archive.apache.org/dist/incubator/pulsar/pulsar-{{ pulsar_version }}/apache-pulsar-{{ pulsar_version }}-bin.tar.gz
+        src: https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=pulsar/pulsar-{{ pulsar_version }}/apache-pulsar-{{ pulsar_version }}-bin.tar.gz
```
Comment (sijie) [inline deployment/terraform-ansible/deploy-pulsar.yaml:45]: This only works for the latest version. It doesn't work for old versions. Did you encounter any troubles using the archive path?

Comment (kristi) [reply deployment/terraform-ansible/deploy-pulsar.yaml:45]: Oh, thanks for testing this. I eventually hit the archive daily limit after testing a bunch. I can see if there is a mirror url for other versions, or else I'll revert to the archive url.

File: site2/docs/deploy-aws.md [hunk 3]
```diff
@@ -173,7 +173,11 @@ Remember to enter this command just only once. If you attempt to enter this comm
 
 ## Run the Pulsar playbook
 
-Once you have created the necessary AWS resources using Terraform, you can install and run Pulsar on the Terraform-created EC2 instances using Ansible. To do so, enter this command:
+Once you have created the necessary AWS resources using Terraform, you can install and run Pulsar on the Terraform-created EC2 instances using Ansible. 
+
+(Optional) If you want to use any [builtin IO connectors](io-connectors.md) , edit the `Download Pulsar IO packages` task in the `deploy-pulsar.yaml` file and uncomment the connectors you want to use.
```
Comment (Huanli-Meng) [inline site2/docs/deploy-aws.md:178]:
```suggestion
(Optional) If you want to use any [built-in IO connectors](io-connectors.md) , edit the `Download Pulsar IO packages` task in the `deploy-pulsar.yaml` file and uncomment the connectors you want to use. 
```

**Comment by Huanli-Meng** [comment]
@jiazhai , @sijie, I think a doc-required label should be added as some config files are updated. we need to update them to pulsar doc. i have no right to add a label, please help add. Thanks.

**Comment by sijie** [comment]
@Huanli-Meng done

**Comment by wolfstudy** [comment]
Move this change to 2.6.2, because the #8310 depends on it.

**Pull request merged**